src/tables.jl

```diff
@@ -47,3 +47,585 @@ function schema_names(featuredefn::IFeatureDefnView)
+        geomcols,
+        fieldtypes,
+        string.(sch.names),
+    )
+
+    # Infer geometry and field types
+    AGtypes = _infergeometryorfieldtypes(sch, rows, spgeomcols, spfieldtypes)
+
+    # Create layer
+    (layer, geomindices, fieldindices) = _create_empty_layer_from_AGtypes(
+        string.(sch.names),
+        AGtypes,
+        layer_name,
+    )
+
+    # Populate layer
+    for row in rows
+        rowvalues =
+            [Tables.getcolumn(row, col) for col in Tables.columnnames(row)]
+        rowgeoms = view(rowvalues, geomindices)
+        rowfields = view(rowvalues, fieldindices)
+        addfeature(layer) do feature
+            # For geometry fields both `missing` and `nothing` map to not geometry set
+            # since in GDAL <= v"3.3.2", special fields as geometry field cannot be NULL
+            # cf. `OGRFeature::IsFieldNull( int iField )` implemetation
+            for (j, val) in enumerate(rowgeoms)
+                if val !== missing && val !== nothing
+                    setgeom!(feature, j - 1, convert(IGeometry, val))
+                end
+            end
+            for (j, val) in enumerate(rowfields)
+                if val === missing
+                    setfieldnull!(feature, j - 1)
+                elseif val !== nothing
+                    setfield!(feature, j - 1, val)
+                end
+            end
+        end
+    end
+
+    return layer
+end
+
+"""
+    _fromtable(::Tables.Schema{names,nothing}, rows; name::String = "")
+
+Handles the case where types in schema is `nothing`
+
+# Implementation
+Tables.Schema types are extracted from `rows`'s columns element types before calling `_fromtable(Tables.Schema(names, types), rows; name = name)`
+
+"""
+function _fromtable(
+    ::Tables.Schema{names,nothing},
+    rows;
+    kwargs...,
+)::IFeatureLayer where {names}
+    cols = Tables.columns(rows)
+    types = (eltype(collect(col)) for col in cols)
+    return _fromtable(Tables.Schema(names, types), rows; kwargs...)
+end
+
+"""
+    _fromtable(::Tables.Schema{names,nothing}, rows; name::String = "")
+
+Handles the case where schema is `nothing`
+
+# Implementation
+Tables.Schema names are extracted from `rows`'s columns names before calling `_fromtable(Tables.Schema(names, types), rows; name = name)`
+
+"""
+function _fromtable(::Nothing, rows; kwargs...)::IFeatureLayer
+    state = iterate(rows)
+    state === nothing && return IFeatureLayer()
+    row, _ = state
+    names = Tables.columnnames(row)
+    return _fromtable(Tables.Schema(names, nothing), rows; kwargs...)
+end
+
+"""
+    IFeatureLayer(table; kwargs...)
+
+Construct an IFeatureLayer from a source implementing Tables.jl interface
+
+## Keyword arguments
+- `layer_name::String = ""`: name of the layer
+- `geomcols::Union{Nothing, Vector{String}, Vector{Int}} = nothing`: if `geomcols` is different from nothing, only the specified columns (by names or number) will be converted to geomfields
```

Comment:
Have we considered not having this as part of the function?

* Rationale: If users knew what the `geomcols` are, they can convert before calling this function.
* In this proposal: the `fieldtypes` keyword arg should be the only way in which users can hint/coerce. 
* This might avoid the need to have a `_coherencecheckandnormalizationofkwargs(...)`

Reply:
> Rationale: If users knew what the `geomcols` are, they can convert before calling this function.

It will always be faster for the user not to have to convert :-)

---

> In this proposal: the `fieldtypes` keyword arg should be the only way in which users can hint/coerce.
It is a lot easier:
- to type a list of `Int` or `String` for `geomcols` than 
- to type in a `Dict` a list of `Int` or `String` PLUS a list `IGeometry{OGRwkbGeometryType}` or `Tuple{OGRFieldType, OGRFieldSubType}`  for `fieldtypes` kwarg

Therefore I would rather drop the `fieldtypes` kwarg.

---

> This might avoid the need to have a `_coherencecheckandnormalizationofkwargs(...)`

You would still need part of what is done in this function either you drop `geomcols` or `fieldtypes`.
But the function would be simpler if you drop `fieldtypes` than if you drop `geomcols
There 4 parts in it (gathering the comments within) :
| Part | Description | `fieldtypes`dropped | `geomcols` dropped |
|---|:--|---|---|
|1|Test coherence of `geomcols` and normalize it with indices of schema names |kept|dropped|
|2|Test coherence `fieldtypes` with schema names, and normalize it to a `Dict{Int, ...}` with indices of schema names|dropped|kept|
|3|Test coherence of `spfieldtypes` and `spgeomcols`|dropped|dropped|
|4|Test coherence of `OGRFieldType` and `OGRFieldSubType` in `fieldtypes` kwarg|dropped|kept|

So it is 3 drops for `fieldtypes` dropping against 2 drops for `geomcols` dropping: `geomcols` dropping wins :-)